### PR TITLE
Signup

### PR DESCRIPTION
**Pull request by zuzanna81**
### Description:
- Added feature and scenarios for signup feature
- Added step definitions for signup feature
- Added rspec
- Added devise for authentication

### Things we've learned
- How to implement and use devises

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -16,6 +16,7 @@ gem 'jbuilder', '~> 2.5'
 
 
 group :development, :test do
+  gem 'devise'
```
Comment (tochman) [inline Gemfile]: This gem should go in the production group

Comment (zuzanna81) [reply Gemfile]: Shall we put that gem in 'general' group then?

File: features/signup.feature [hunk 2]
```diff
@@ -7,45 +7,72 @@ Feature: Visitor Sign Up
...
+  Scenario: Visitor fills in all fields with taken email [Sad Path]
+    When I fill in "Email" with "cutie123@hotmail.com"
+    When I fill in "Password" with "whatever"
+    And I fill in "Password confirmation" with "whatever"
+    And I click on the "Sign up" button or link
+    Then I should see "1 error prohibited this user from being saved:"
+    And I should see "Email has already been taken"
+
+
+  Scenario: Visitor does not fill in any fields [Sad Path]
```
Comment (SophieSjoberg) [inline features/signup.feature:73]: The only thing - I should probably divide email and password errors into two scenarios. But maybe that's just a personal preference. :)

File: features/step_definitions/signup_steps.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+Given("the following user exists") do |table|
+  table.hashes.each do |user|
+    User.create(user)
```
Comment (tochman) [inline features/step_definitions/signup_steps.rb]: Please make use of FactoryBot

File: config/routes.rb [hunk 4]
```diff
@@ -1,4 +1,11 @@
 Rails.application.routes.draw do
 
+  devise_for :users
+  get 'users/show'
+
+  get 'users/create'
+
+  get 'users/new'
```
Comment (tochman) [inline config/routes.rb]: Please remove these routes

File: features/signup.feature [hunk 5]
```diff
@@ -0,0 +1,78 @@
+Feature: Visitor Sign Up
+  As a user
+  In order to register an account
+  I need to be able to write in my user details and register
+
+  Background:
+    Given I visit the "Landing" page
+    When I click on the "Sign up" button or link
```
Comment (tochman) [inline features/signup.feature]: Why not just go directly to the Sign up page?

File: features/signup.feature [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    Then I should be redirected to the "Sign up" page
+    Given the following user exists
+      | email                  | password    | password_confirmation |
+      | cutie123@hotmail.com   | hello123    | hello123              |
```
Comment (tochman) [inline features/signup.feature]: Please move this to the top of the background

File: features/step_definitions/signup_steps.rb (hunk 3 above)
Comment (tochman) [reply features/step_definitions/signup_steps.rb]: Please make use of FactoryBot

File: spec/models/user_spec.rb [hunk 7]
```diff
@@ -0,0 +1,16 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'DB Table' do
```
Comment (tochman) [inline spec/models/user_spec.rb:4]: Please write a spec for a working factory

**Review by tochman — DISMISSED** [review]
Take a break and Mr R will come in and help you clean up this PR

File: spec/models/user_spec.rb (hunk 7 above)
Comment (diraulo) [reply spec/models/user_spec.rb:4]: There's now a spec for a valid factory

File: features/signup.feature (hunk 2 above)
Comment (diraulo) [reply features/signup.feature:73]: Not a biggy.. 😉

**Review by diraulo — DISMISSED** [review]
Looks good to me. What do you think now @tochman

**Review by ScottyyyB — DISMISSED** [review]

**Review by ScottyyyB — APPROVED** [review]

**Pull request merged**